### DISCUSSION TITLE
Overlay png over gif at the first frame

### DISCUSSION THREAD

**Discussion by AquaCylinder** [ImageMagick/ImageMagick#2241] [Help] [answered]
Hi again,
i'm trying to overlay a png on top of a gif i did it but it's not perfect as the overlayed png appears on the second frme instead of the very moment the gif starts. i used this: `convert -dispose none On-Normal-Segment.png Rainbow-Standard-Blank-NA.gif -delay 1x30 -loop 0 Rainbow-Standard-Blank-On.gif` code to get this: 
![Rainbow-Standard-Blank-On](https://user-images.githubusercontent.com/56627792/86627085-ff3ff480-bfc7-11ea-8cf7-a502dfbeb4c2.gif)
But as you can see is the png appearing on the second frame, is there a way to make it so the png is there from the start?
Thanks i love this community!

**Comment by fmw42** [comment, marked as answer]
Please post your two input images (so we can properly test a suggested command) and tell us what your IM version and platform are. I think you need to use -layers composite to do that. Try

`convert -delay 1x30 Rainbow-Standard-Blank-NA.gif -coalesce null: On-Normal-Segment.png -layers composite -layers optimize -loop 0 Rainbow-Standard-Blank-On.gif`

**Reply by AquaCylinder** [reply]
![Rainbow-Standard-Blank-NA](https://user-images.githubusercontent.com/56627792/86630538-3fee3c80-bfcd-11ea-8055-69534f0e46b5.gif)![On-Normal-Segment](https://user-images.githubusercontent.com/56627792/86630544-41b80000-bfcd-11ea-805b-60bbcc0b5b2d.png)
those are the input images and i modified your awnser a bit so i removed the `-layers optimize` as that gives me very bad edges
![Rainbow-Standard-Blank-Onbad](https://user-images.githubusercontent.com/56627792/86631291-37e2cc80-bfce-11ea-823a-01b1fa88954b.gif)
 but other than that it's perfect! so i have IM version 7.0.10-19 Q16 and i'm on windows 10. so my final code would be: 
`magick -delay 1x30 Rainbow-Standard-Blank-NA.gif -coalesce null: On-Normal-Segment.png -layers composite -loop 0 Rainbow-Standard-Blank-On.gif` With this as a output:
![Rainbow-Standard-Blank-On](https://user-images.githubusercontent.com/56627792/86631356-51841400-bfce-11ea-9947-1d47dc9b2eb1.gif)
(this is just so people have a simple awnser in the future).
So it works now! Thank you!

**Reply by fmw42** [reply]
If you are on IM 7, then you should use magick in place of convert in your command.

**Reply by AquaCylinder** [reply]
oh thank you i'll change that right away! Another quick question i don't think there is a way to remove 2 frames every 3 frames? like have on-off-off-on-off-off-on-off-off converted to on-on-on right? just curious

**Comment by snibgo** [comment]
> Another quick question i don't think there is a way to remove 2 frames every 3 frames?

I bumped into an option that might do that, possibly a "-define frames:" option, perhaps on the page https://www.imagemagick.org/script/defines.php , but that page seems currently unavailable.

**Reply by fmw42** [reply]
That page is at https://imagemagick.org/script/defines.php

`-define frames:step`

It is very new in ImageMagick 7 (only)

Try setting a step of 3 and see if that does what you want.

**Reply by AquaCylinder** [reply]
I'll have a look at that thank you!
